plugins/file_lumberdash/lib/src/file_lumberdash.dart

```diff
@@ -0,0 +1,73 @@
+import 'dart:io';
+
+import 'package:lumberdash/lumberdash.dart';
+import 'package:synchronized/synchronized.dart';
+import 'package:meta/meta.dart';
+
+/// [LumberdashClient] that writes your logs to the given directory path
+/// in the file system
+class FileLumberdash extends LumberdashClient {
+  final String _filePath;
+  static final _lock = Lock();
+
+  FileLumberdash({
+    @required String filePath,
+  })  : assert(filePath != null),
+        _filePath = filePath;
+
+  /// Records a regular message
+  @override
+  void logMessage(String message, [Map<String, String> extras]) {
+    if (extras != null) {
+      _log('[MESSAGE] $message, extras: $extras');
+    } else {
+      _log('[MESSAGE] $message');
+    }
+  }
+
+  /// Records a warning message
+  @override
+  void logWarning(String message, [Map<String, String> extras]) {
+    if (extras != null) {
+      _log('[WARNING] $message, extras: $extras');
+    } else {
+      _log('[WARNING] $message');
+    }
+  }
+
+  /// Records a fatal message
+  @override
+  void logFatal(String message, [Map<String, String> extras]) {
+    if (extras != null) {
+      _log('[FATAL] $message, extras: $extras');
+    } else {
+      _log('[FATAL] $message');
+    }
+  }
+
+  /// Records an error message
+  @override
+  Future<void> logError(exception, [dynamic stacktrace]) async {
+    if (stacktrace != null) {
+      _log('[ERROR] { exception: $exception, stacktrace: $stacktrace }');
+    } else {
+      _log('[ERROR] { exception: $exception }');
+    }
+  }
+
+  Future<void> _log(String data) async {
+    try {
+      final file = File(_filePath);
```

Comment:
couldn't this be a private member variable of the class since the `_filePath` is final?